ci_framework/roles/operator_build/README.md

```diff
@@ -0,0 +1,58 @@
+## Role: operator_build
+Builds OpenStack k8s operators and push them to a registry.
+When building an operator from a Pull Request, it is mandatory to provide the PR Owner and the PR SHA *only* if
+you want to build meta-operator too, so the role can properly replace api references in meta-operator.
+
+## Parameters
+* `cifmw_operator_build_org`: (String) Operator's organization on GitHub. Defaults to **openstack-k8s-operators**.
+* `cifmw_operator_build_push_registry`: (String) Registry used to push images. Defaults to **quay.rdoproject.org**.
+* `cifmw_operator_build_push_org`: (String) Registry's organization to push image to. Defaults to **openstack-k8s-operators**.
+* `cifmw_operator_build_operators`: (List) List of operators to build (it should not include meta-operator). Defaults to **[]**.
+* `cifmw_operator_build_meta_name`: (String) Meta operator's name. Defaults to **openstack-operator**.
+* `cifmw_operator_build_meta_src`: (String) Directory with src code for meta operator. Defaults to **"{{ ansible_user_dir }}/src/github.com/{{ cifmw_operator_build_org }}/{{ cifmw_operator_build_meta_name }}"**
+* `cifmw_operator_build_meta_build`: (Boolean) When set to **true** updates meta-operator's go.mod when build operators and builds meta-operator in the end. Default to **true**.
+
+## TODO
+* Include PR Owner and PR SHA for meta-operator
+* Include tasks to get PR Owner and PR SHA info from **zuul** dict.
+* Include tasks to get PR Owner and PR SHA info from **Prow** environment.
+* Add molecule tests.
+
+## Examples
+### 1 - Building mariadb-operator checked-out from a PR and meta-operator
+  ```yaml
+- hosts: all
+  tasks:
+    - name: Builds mariadb-operator and meta-operator
+      include_role:
+        name: operator_build
+      vars:
+        cifmw_operator_build_push_registry: "quay.io"
+        cifmw_operator_build_push_org: "myorg"
+        cifmw_operator_build_meta_build: true
+        cifmw_operator_build_meta_src: "/home/user/openstack-operator"
+        cifmw_operator_build_operators:
+          - name: "mariadb-operator"
+            src: "/home/user/mariadb-operator"
+            pr_owner: "ci-framework/openstack-operator"
```

Comment:
In Zuul, the idea is that we can go through zuul inventory, find all `openstack-k8s-operators` repos, and populate this list of operators. We can get all those info from zuul. (We can get PR number, and with that get PR Owner using GH API)[1]

[1] https://github.com/rdo-infra/review.rdoproject.org-config/blob/142a98841377483222c7e331754f3436cd6ac3b6/playbooks/crc/build_operators.yaml#L8-L34